presto-verifier/src/main/java/com/facebook/presto/verifier/Verifier.java

```diff
@@ -176,6 +184,64 @@ public int run(List<QueryPair> queries)
         return failed;
     }
 
+    private int getDefultHashPartitionCount()
+    {
+        String url = config.getTestGateway();
+        String username = Optional.ofNullable(config.getTestUsernameOverride()).orElse("verifier-test");
+        String password = Optional.ofNullable(config.getTestPasswordOverride()).orElse(null);
+
+        int defaultHashPartitionCount = 0;
+        int numOfActiveNodes = 0;
+        try {
+            ResultSet result = executeQuery(url, username, password, "select count(*) as NUM_OF_WORKERS from system.runtime.nodes where coordinator=false");
+            while (result.next()) {
+                numOfActiveNodes = result.getInt("NUM_OF_WORKERS");
+            }
+            result.close();
+
+            result = executeQuery(url, username, password, "show session");
+            while (result.next()) {
+                if ((result.getString("Name")).equals("hash_partition_count")) {
+                    defaultHashPartitionCount = Integer.parseInt(result.getString("Value"));
+                    break;
+                }
+            }
+            result.close();
+        }
+        catch (SQLException e) {
+            e.printStackTrace();
+        }
+
+        return Math.min(defaultHashPartitionCount, numOfActiveNodes);
+    }
+
+    private ResultSet executeQuery(String url, String username, String password, String sql)
+    {
+        try (Connection connection = DriverManager.getConnection(url, username, password)) {
+            try (Statement statement = connection.createStatement()) {
+                try {
+                    ResultSet result = statement.executeQuery(sql);
+                    return result;
+                }
+                catch (SQLException e) {
+                    throw e;
+                }
+                catch (Exception e) {
+                    throw Throwables.propagate(e);
+                }
+            }
+        }
+        catch (SQLException e) {
+            Exception exception = e;
+            if (("Error executing query".equals(e.getMessage()) || "Error fetching results".equals(e.getMessage())) &&
+                    (e.getCause() instanceof Exception)) {
+                exception = (Exception) e.getCause();
+            }
+            exception.printStackTrace();
+            return null;
```

Comment:
Since you are not handling null `ResultSet` in the caller of this function, it would be weird if we see the `NullPointerException` in the log that's actually caused by a `SQLException`.

Maybe you just need to propagate the exception out.